mmlearn/modules/encoders/text.py

```diff
@@ -171,3 +195,108 @@ def forward(self, inputs: Dict[str, Any]) -> BaseModelOutput:
             hidden_states=outputs.hidden_states,
             attentions=outputs.attentions,
         )
+
+
+@store(group="modules/encoders", provider="mmlearn", hydra_convert="object")
+class Data2VecTextEncoder(TextEncoderBase):
+    """Text encoder for Data2Vec implementation."""
+
+    def __init__(
+        self,
+        model_name_or_path: str,
+        head_layers: int = 1,
+        pretrained: bool = True,
+        freeze_layers: Union[int, float, List[int], bool] = False,
+        freeze_layer_norm: bool = True,
+        peft_config: Optional["PeftConfig"] = None,
+        model_kwargs: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        """Initialize the encoder."""
+        super().__init__()
+
+        if model_kwargs is None:
+            model_kwargs = {}
+        model_kwargs["output_hidden_states"] = True
+        model_kwargs["add_pooling_layer"] = False
+
+        # Load base model
+        self.model = (
+            AutoModel.from_pretrained(model_name_or_path, **model_kwargs)
+            if pretrained
+            else AutoModel.from_config(
+                AutoModel.config_class.from_pretrained(model_name_or_path)
+            )
```

Comment:
The utility function for loading huggingface models should be able to do this as well.

Reply:
Right, missed that